### PR TITLE
[text analytics] return None for offset and length for v3.0

### LINKED ISSUE #13365: [text analytics] make sure v3.0 models don't have offset and length

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #13365

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py [hunk 1]
```diff
@@ -223,12 +227,19 @@ def __init__(self, **kwargs):
 
     @classmethod
     def _from_generated(cls, entity):
+        offset = entity.offset
+        length = entity.length
+        if isinstance(entity, _v3_0_models.Entity):
+            # we do not return offset and length for v3.0 since
+            # the correct encoding was not introduced for v3.0
+            offset = None
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py:235]: is this the pattern other libraries have taken? I am new to this paradigm as in TA and FR we can only support one version of the SDK

Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py:235]: for python, we're supposed to support all API versions, the user only needs to pass in the api_version as a kwarg if they don't want the default.

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_linked_entities.py [hunk 2]
```diff
@@ -545,3 +546,36 @@ def callback(pipeline_response, deserialized, _):
...
+    def test_no_offset_length_v3_linked_entity_match(self, client):
+        result = client.recognize_linked_entities(["Microsoft was founded by Bill Gates and Paul Allen"])
+        entities = result[0].entities
+
+        self.assertIsNone(entities[0].matches[0].offset)
+        self.assertIsNone(entities[0].matches[0].length)
+        self.assertIsNone(entities[0].matches[0].offset)
+        self.assertIsNone(entities[0].matches[0].length)
+        self.assertIsNone(entities[0].matches[0].offset)
+        self.assertIsNone(entities[0].matches[0].length)
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_linked_entities.py:581]: the indices are all the same here

Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_linked_entities.py:581]: oof thanks

**Pull request merged**